pkg/operator/ceph/cluster/osd/status.go

```diff
@@ -185,7 +193,20 @@ func (c *Cluster) completeOSDsForAllNodes(config *provisionConfig, configOSDs bo
 				if !ok {
 					logger.Infof("orchestration status config map result channel closed, will restart watch.")
 					w.Stop()
-					<-time.After(100 * time.Millisecond)
+					<-time.After(5 * time.Second)
+					opts.ResourceVersion = ""
+					opts.Watch = false
+					leftNodes := 0
+					leftNodes, remainingNodes, completed, statuses, err = c.checkNodesCompleted(opts, config, configOSDs)
```

Comment:
Instead of `opts`, how about if you passed the `selector`? Then you could construct an independent `opts` struct and you wouldn't have to reset the fields that you need for watching.